server/text_generation_server/models/causal_lm.py

```diff
@@ -654,10 +654,17 @@ def __init__(
             "return_dict": True,
         }
 
-        if model.config.model_type == "llama":
+        if model.config.model_type in ["llama", "mistral"]:
             kwargs["attn_softmax_bf16"] = True
             kwargs["trim_logits"] = True
 
+        if os.getenv("USE_FLASH_ATTENTION", "false").lower() == "true":
+            kwargs["use_flash_attention"] = True
+        if os.getenv("FLASH_ATTENTION_RECOMPUTE", "false").lower() == "true":
+            kwargs["flash_attention_recompute"] = True
+        if os.getenv("FLASH_ATTENTION_CAUSAL_MASK", "false").lower() == "true":
+            kwargs["flash_attention_causal_mask"] = True
+
```

Comment:
Are these params model-specific?

Reply:
Currently it's used for Llama and Mistral.

Reply:
Wouldn't it be easier to add a CLI arg `--use_flash_attention` than using various env variables?

Reply:
I can do either way, not sure what the rule is through here.
I see a lot of them are configured through environment variables though, thought that this would be better since this will be enabled for a couple specific models in a specific cases (long sentences)?
https://github.com/huggingface/tgi-gaudi/blob/habana-main/README.md#environment-variables

Reply:
@kdamaszk Have we added the flash attention args for Llama? I don't find anything in `causal_lm.py` but maybe it is elsewhere.

Reply:
No, I didn't test flash attention in tgi-gaudi at all.

Reply:
Okay, in that case let's keep the env variables to enable flash attention.